plugins/view-transitions/includes/settings.php

```diff
@@ -0,0 +1,402 @@
+			</p>
+			<?php
+		},
+		'reading',
+		array(
+			'before_section' => '<div id="view-transitions">',
+			'after_section'  => '</div>',
+		)
+	);
+
+	$reset_explanation = __( 'To reset, clear the field and save the changes.', 'view-transitions' );
+
+	$fields = array(
+		'default_transition_animation' => array(
+			'title'       => __( 'Default Transition Animation', 'view-transitions' ),
+			'description' => __( 'Choose the animation that is used for the default view transition type.', 'view-transitions' ),
+		),
+		'header_selector'              => array(
+			'title'       => __( 'Header Selector', 'view-transitions' ),
+			'description' => __( 'Provide the CSS selector to detect the global header element.', 'view-transitions' ) . ' ' . $reset_explanation,
+		),
+		'main_selector'                => array(
+			'title'       => __( 'Main Selector', 'view-transitions' ),
+			'description' => __( 'Provide the CSS selector to detect the global main element.', 'view-transitions' ) . ' ' . $reset_explanation,
+		),
+		'post_title_selector'          => array(
+			'title'       => __( 'Post Title Selector', 'view-transitions' ),
+			'description' => __( 'Provide the CSS selector to detect the post title element.', 'view-transitions' ) . ' ' . $reset_explanation,
+		),
+		'post_thumbnail_selector'      => array(
+			'title'       => __( 'Post Thumbnail Selector', 'view-transitions' ),
+			'description' => __( 'Provide the CSS selector to detect the post thumbnail element.', 'view-transitions' ) . ' ' . $reset_explanation,
+		),
+		'post_content_selector'        => array(
+			'title'       => __( 'Post Content Selector', 'view-transitions' ),
+			'description' => __( 'Provide the CSS selector to detect the post content element.', 'view-transitions' ) . ' ' . $reset_explanation,
+		),
+	);
+	foreach ( $fields as $slug => $args ) {
+		add_settings_field(
+			"plvt_view_transitions_{$slug}",
+			$args['title'],
+			'plvt_render_settings_field',
+			'reading',
+			'plvt_view_transitions',
+			array_merge(
+				array(
+					'field'     => $slug,
+					'label_for' => "plvt-view-transitions-field-{$slug}",
+				),
+				$args
+			)
+		);
+	}
+}
+
+/**
+ * Renders a settings field for the View Transitions configuration.
+ *
+ * @since 1.0.0
+ * @access private
+ *
+ * @param array{ field: non-empty-string, title: non-empty-string, description: string, label_for: non-empty-string } $args {
+ *     Associative array of arguments.
+ *
+ *     @type string $field       The slug of the sub setting controlled by the field.
+ *     @type string $title       The title for the field.
+ *     @type string $description Optional. A description to show for the field.
+ *     @type string $label_for   ID to use for the field control.
+ * }
+ */
+function plvt_render_settings_field( array $args ): void {
+	$option = plvt_get_stored_setting_value();
+
+	switch ( $args['field'] ) {
```

Comment:
There's also no guarantee that `$args['field']` is defined. If not set, perhaps this should short-circuit if the expected array keys aren't defined?

```suggestion
	if ( ! isset( $args['field'], $args['label_for'], $args['field'], $args['description'] ) ) {
		return;
	}
	switch ( $args['field'] ) {
```

Granted, this is highly unlikely.

Reply:
There's no guarantee from a PHP perspective, but isn't it required at least from a PHPStan perspective due to how the `@param` is defined? Also it's worth noting this is a private function, so that plus the PHPStan annotation might make this unnecessary.

On the other hand, of course it's good for defensive coding. In that case though, I think we should only bail for `field` and `label_for` missing, since `description` is optional in the first place - if it's not set, we should just not render a description.